app/assets/javascripts/modules/FirmMap.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * # FirmMap
+ *
+ * A map that plots Advisers and Offices.
+ *
+ * @module FirmMap
+ * @returns {class} FirmMap
+ */
+
+define(['jquery', 'DoughBaseComponent'],
+       function($, DoughBaseComponent) {
+  'use strict';
+
+  var FirmMapProto,
+      defaultConfig = {};
+
+  /**
+   * @constructor
+   * @extends {DoughBaseComponent}
+   * @param {HTMLElement} $el    Element with dough-component on it
+   * @param {Object}             config Hash of config options
+   */
+  function FirmMap($el, config) {
+    FirmMap.baseConstructor.call(this, $el, config, defaultConfig);
+  }
+
+  /**
+   * Inherit from base module, for shared methods and interface
+   */
+  DoughBaseComponent.extend(FirmMap);
+  FirmMap.componentName = 'FirmMap';
+  FirmMapProto = FirmMap.prototype;
+
+  /**
+   * Init function
+   *
+   * Set up listeners and accept promise
+   *
+   * @param {Object} initialised Promise passed from eventsWithPromises (RSVP Promise).
+   * @returns {MultiTableFilter}
+   */
+  FirmMapProto.init = function(initialised) {
+    this.initializeGoogleMaps();
+    this._initialisedSuccess(initialised);
+    return this;
+  };
+
+  /**
+   * initializeGoogleMaps
+   *
+   * Adds the google maps script tag, and adds setupMap to the window object
+   * so the library can find it.
+   */
+  FirmMapProto.initializeGoogleMaps = function() {
+    window.doughMap__initMap = $.proxy(this.setupMap, this);
+    var $map = this._getMapElement(),
+        apiKey = $map.data('dough-map-key');
+    $map.parent().append('<script defer src="https://maps.googleapis.com/maps/api/js?key=' +
```

Comment:
The readme for `requirejs-plugins` suggested that `async` was for google maps, so I went with that.